autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/ScheduleListItem/Buttons/DeleteScheduleButton.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Button, Dialog, DialogTitle, DialogContent, DialogContentText, DialogActions, IconButton, Tooltip,
+} from '@material-ui/core';
+import RemoveIcon from '@material-ui/icons/Delete';
+import { removeSchedule, unsaveSchedule } from '../../../../../redux/actions/schedules';
+import { RootState } from '../../../../../redux/reducer';
+import Meeting from '../../../../../types/Meeting';
+
+interface DeleteScheduleButtonProps {
+  index: number;
+}
+
+const DeleteScheduleButton: React.FC<DeleteScheduleButtonProps> = ({ index }) => {
+  const dispatch = useDispatch();
+
+  const schedule = useSelector<RootState, Meeting[]>((state) => (
+    state.schedules.allSchedules[index]
+  ));
+  const savedSchedules = useSelector<RootState, Meeting[][]>((state) => (
+    state.schedules.savedSchedules
```

Comment:
I think I recall you saying that you had changed to making `saved` a field for each schedule rather than storing a separate array in Redux. Was that in a different PR?

Reply:
Yeah that's in the rename schedules PR, I decided to make all the redux changes there so it's more clear what exactly I changed for it